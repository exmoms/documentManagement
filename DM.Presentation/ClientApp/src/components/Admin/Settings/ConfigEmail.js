/**
 * @file Contains ConfigEmail component implementation
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import {
  Button,
  Container,
  CssBaseline,
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  TextField,
} from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import clsx from "clsx";
import React, { Component } from "react";
import { withTranslation } from "react-i18next";
import { postDataToAPI } from "../../../api/PostData";
import { fetchData } from "../../../api/FetchData";
import { Notification } from "../../Admin/Notifications";

const useStyles = (theme) => ({
  paper: {
    marginTop: theme.spacing(4),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 1, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
});

class ConfigEmail extends Component {
  t = this.props["t"];

  addConfigURL = "/api/Mail/AddConfig";
  testMailServerConnectionURL = "/api/Mail/TestConnection";
  getConfigURL = "/api/Mail/GetConfig";
  buttonCase = "";

  constructor(props) {
    super(props);
    this.state = {
      password: "",
      showPassword: false,
      portError: false,
      portErrorText: "",
      testedSuccess: false,
      open: false,
      error: false,
      errorMessage: [],
    };
  }

  componentDidMount() {
    fetchData(this.getConfigURL).then((data) => {
      document.getElementById("email").value = data.email;
      document.getElementById("server").value = data.server;
      document.getElementById("port").value = data.port;
      this.setState({
        password: data.password,
      });
    });
  }

  handlePassword = (e) => {
    this.setState({ password: e.target.value });
  };

  handleClickShowPassword = () => {
    this.setState({ showPassword: !this.state.showPassword });
  };

  handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  /**
   * handle onChange event for port to check if it's valid
   */
  onChangePort = (e) => {
    const port = e.target.value;

    const validPort =
      port &&
      /^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$/i.test(
        port
      );

    if (!validPort) {
      this.setState({
        portError: true,
        portErrorText: this.t("invalid_port"),
      });
    } else {
      this.setState({
        portError: false,
        portErrorText: "",
      });
    }
  };

  handelSubmit = (e) => {
    e.preventDefault();

    const email = document.getElementById("email").value;
    const server = document.getElementById("server").value;
    const port = document.getElementById("port").value;

    const formData = {
      Server: server,
      Port: parseInt(port),
      email: email,
      Password: this.state.password,
    };

    if (
      email !== "" ||
      server !== "" ||
      port !== "" ||
      this.state.password !== ""
    ) {
      switch (this.buttonCase) {
        case "setup":
          if (this.state.testedSuccess) {
            postDataToAPI(this.addConfigURL, formData).then((addRes) => {
              if (addRes.status === 200) {
                this.setState({ open: true, error: false, errorMessage: [] });
                // alert(`Mail configuration is added successfully.`);
              } else {
                this.setState({
                  open: true,
                  error: true,
                  errorMessage: [this.t("Mail_con_err")],
                });
                // alert(`ERROR ${addRes.status}: ${addRes.statusText}`);
              }
            });
          } else {
            // alert("You have to test your configuration and be sure it works.");
            this.setState({
              open: true,
              error: true,
              errorMessage: [this.t("Mail_con_test")],
            });
          }
          break;

        case "test":
          postDataToAPI(this.testMailServerConnectionURL, formData).then(
            (res) => {
              if (res.status === 200) {
                // alert(
                //   `Mail configuration is tested and it successfully works.`
                // );
                this.setState({
                  testedSuccess: true,
                  open: true,
                  error: false,
                });
              } else {
                // alert(`ERROR ${res.status}: ${res.statusText}`);
                this.setState({
                  open: true,
                  error: true,
                  errorMessage: [this.t("Mail_con_test_err")],
                });
              }
            },
            (error) => console.log(error)
          );
          break;

        default:
          console.log(this.buttonCase);
          break;
      }
    }
  };

  setOpen = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes } = this.props;
    const { showPassword, password, portErrorText, portError } = this.state;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <form
            className={classes.form}
            onSubmit={this.handelSubmit}
            id="submitForm"
          >
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              name="email"
              label={this.t("emailaddress")}
              defaultValue="Enter your email"
              type="email"
              id="email"
            />
            <FormControl
              id="password-form"
              className={clsx(classes.margin, classes.textField)}
              variant="outlined"
            >
              <InputLabel
                htmlFor="outlined-adornment-password"
                id="password-label"
              >
                {this.t("password")}
              </InputLabel>
              <OutlinedInput
                id="OutlinedInput"
                name="password"
                type={showPassword ? "text" : "password"}
                onChange={this.handlePassword}
                value={password}
                endAdornment={
                  <InputAdornment position="end" id="input-adornment-password">
                    <IconButton
                      aria-label="toggle password visibility"
                      onClick={this.handleClickShowPassword}
                      onMouseDown={this.handleMouseDownPassword}
                      edge="end"
                    >
                      {showPassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                }
                labelWidth={70}
              />
            </FormControl>
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              name="server"
              label={this.t("server")}
              defaultValue="Enter server of email"
              type="text"
              id="server"
            />
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              onChange={this.onChangePort}
              error={portError}
              helperText={portErrorText}
              name="port"
              label={this.t("port")}
              defaultValue={-1}
              type="number"
              id="port"
            />
            <Button
              id="buttonTest"
              type="submit"
              onClick={() => (this.buttonCase = "test")}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              {this.t("test")}
            </Button>
            <Button
              id="buttonSetUP"
              type="submit"
              onClick={() => (this.buttonCase = "setup")}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              {this.t("setup_email_configuration")}
            </Button>
          </form>
        </div>
        <Notification
          id="notification"
          open={this.state.open}
          setOpen={this.setOpen}
          error={this.state.error}
          errorMessage={this.state.errorMessage}
        />
      </Container>
    );
  }
}

export default withStyles(useStyles)(withTranslation()(ConfigEmail));
