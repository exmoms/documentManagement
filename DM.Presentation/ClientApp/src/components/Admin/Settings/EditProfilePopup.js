/**
 * @file Contains BrowseAggregatedDocsPopUp component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import Paper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableRow from "@material-ui/core/TableRow";
import Typography from "@material-ui/core/Typography";
import React, { Component } from "react";
import { withTranslation } from "react-i18next";
import { fetchData } from "../../../api/FetchData";
import { postDataToAPI } from "../../../api/PostData";
import { Notification } from "../Notifications";
import EditUserProfileByAdmin from "./EditUserProfileByAdmin";

const styles = (theme) => ({
  paper: {
    maxWidth: "auto",
  },
  margin: {
    margin: theme.spacing(2),
  },
});

class EditProfilePopup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: this.props.user,
      userProfile: null,
      errors: {
        confirmErrors: "",
        newPasswordErrors: "",
        submissionErrors: "",
      },
      showSuccess: false,
    };
    this.updateParentState = this.props.action;
  }
  t = this.props["t"];

  componentDidUpdate(prevProps) {
    if (prevProps.user.userId !== this.props.user.userId) {
      fetchData(`/api/user/${this.props.user.userId}`).then((profileData) => {
        this.setState({ userProfile: profileData });
      });
    }
  }

  toggleShow = (show) => {
    this.updateParentState(show);
  };

  handelSubmit = (event) => {
    event.preventDefault();
    let email = document.getElementById("edit-profile-by-admin-email").value;
    let tel = document.getElementById("edit-profile-by-admin-tel").value;
    let newPassword = document.getElementById(
      "edit-profile-by-admin-outlined-adornment-password"
    ).value;
    let confirmPassword = document.getElementById(
      "edit-profile-by-admin-confirmPassword"
    ).value;
    let validPassword = false;
    if (newPassword === "") {
      validPassword = newPassword === confirmPassword;
    } else {
      validPassword =
        newPassword === confirmPassword &&
        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\w]).{8,}$/i.test(newPassword);
    }

    if (validPassword) {
      const formData = {
        userId: this.props.user.userId,
        email: email,
        phoneNumber: tel,
        NewPassword: newPassword,
        ConfirmPassword: confirmPassword,
      };
      postDataToAPI("/api/user/EditUserProfileByAdmin", formData)
        .then(
          (res) => {
            if (res.ok) {
              this.setState({
                errors: {
                  confirmErrors: "",
                  newPasswordErrors: "",
                  submissionErrors: "",
                },
                showSuccess: true,
              });
            }
            return res.json();
          },
          (error) => console.log(error)
        )
        .then(
          (data) => {
            this.setState({
              errors: {
                confirmErrors: "",
                newPasswordErrors: "",
                submissionErrors: data.error,
              },
              showSuccess: true,
            });
          },
          (error) => console.log(error)
        );

      this.toggleShow(false);
    } else {
      this.setState({
        errors: {
          confirmErrors:
            newPassword === confirmPassword
              ? ""
              : this.t("password_mismatchingWithComfirm"),
          newPasswordErrors: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\w]).{8,}$/i.test(
            newPassword
          )
            ? ""
            : this.t("password_formateError"),
        },
      });
    }
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.toggleShow(false)}
          aria-labelledby="form-dialog-title"
          className={classes.paper}
        >
          <DialogTitle id="form-dialog-title">
            <Typography variant="h6" gutterBottom align="left">
              {this.t("edit_user_profile")} : {this.props.user.userName}
            </Typography>
          </DialogTitle>
          <form
            className={classes.form}
            id="edit-profile-form"
            onSubmit={this.handelSubmit}
          >
            <DialogContent>
              <TableContainer component={Paper}>
                <Table className={classes.table} aria-label="simple table">
                  <TableBody>
                    <TableRow key="search-form">
                      <TableCell>
                        <EditUserProfileByAdmin
                          profileData={this.state.userProfile}
                          errors={this.state.errors}
                        />
                      </TableCell>
                    </TableRow>
                  </TableBody>
                </Table>
              </TableContainer>
            </DialogContent>
            <DialogActions>
              <Button
                id="submit-btn"
                type="submit"
                // variant="contained"
                color="primary"
                className={classes.submit}
              >
                {this.t("submit_changes")}
              </Button>
              <Button
                id="CancelButton"
                onClick={() => this.toggleShow(false)}
                color="primary"
              >
                {this.t("button_cancel")}
              </Button>
            </DialogActions>
          </form>
        </Dialog>
        <Notification
          id="notification"
          open={this.state.showSuccess}
          setOpen={(show) => this.setState({ showSuccess: show })}
          error={!!this.state.errors.submissionErrors}
          errorMessage={this.state.errors.submissionErrors}
        />
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(EditProfilePopup));
