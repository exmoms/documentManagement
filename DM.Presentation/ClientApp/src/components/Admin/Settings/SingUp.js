import React, { useState } from "react";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";

import InputLabel from "@material-ui/core/InputLabel";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import { postDataToAPI } from "../../../api/PostData";
import { useTranslation } from "react-i18next";
//import { useAuth } from "../../../context/auth";
import LinearProgress from "@material-ui/core/LinearProgress";
import { Notification } from "../Notifications";

function Copyright() {
  const { t } = useTranslation();
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright Â© "}
      <Link color="inherit" href="https://material-ui.com/">
        {t("link")}
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  formControl: {
    minWidth: "100%",
  },
}));

export default function SignUp(props) {
  const { t } = useTranslation();
  const classes = useStyles();

  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [email, setEmail] = useState("");
  const [role, setRole] = useState("");
  const [loading, setloading] = useState(false);

  const [errorMessage, seterrorMessage] = useState([]);
  const [error, seterror] = useState(false);
  const [open, setOpen] = React.useState(false);

  const regester_file = {
    UserName: userName,
    Email: email,
    Password: password,
    ConfirmPassword: confirmPassword,
    Roles: [
      {
        roleName: "Admin",
        isSelected: role === "Admin" ? true : false,
      },
      {
        roleName: "User",
        isSelected: role === "User" ? true : false,
      },
    ],
  };

  const onchange = (event) => {
    if (event.target.name === "signup-userName") {
      setUserName(event.target.value);
    } else if (event.target.name === "signup-email") {
      setEmail(event.target.value);
    } else if (event.target.name === "signup-password") {
      setPassword(event.target.value);
    } else if (event.target.name === "signup-confirmPassword") {
      setConfirmPassword(event.target.value);
    } else if (event.target.name === "role") {
      setRole(event.target.value);
    }
  };

  const handelSubmit = async (event) => {
    event.preventDefault();

    if (userName.length === 0) {
      seterror(true);
      seterrorMessage([t("userName_error")]);
      setOpen(true);
    } else if (email.indexOf("@") === -1 || email.indexOf(".") === -1) {
      seterror(true);
      seterrorMessage([t("email_error")]);
      setOpen(true);
    } else if (role === "") {
      seterror(true);
      seterrorMessage([t("role_error")]);
      setOpen(true);
    } else if (password.length < 8 || confirmPassword.length > 12) {
      seterror(true);
      seterrorMessage([t("password_charachters_no")]);
      setOpen(true);
    } else if (password !== confirmPassword) {
      seterror(true);
      seterrorMessage([t("password_mismatch")]);
      setOpen(true);
    } else {
      setloading(true);

      let response = await postDataToAPI("/api/user/Register", regester_file);
      if (response.ok) {
        props.updateUserTable(true);
        seterror(false);
        setOpen(true);
        setloading(false);
      } else {
        let json = await response.json();
        seterror(true);
        seterrorMessage(json.error);
        setOpen(true);
        setloading(false);
      }
    }
  };

  const userRole = localStorage.getItem("userRole");

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          {t("signupfornewuser")}
        </Typography>
        <form
          className={classes.form}
          onSubmit={handelSubmit}
          id="submitNewUserForm"
        >
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                onChange={onchange}
                autoComplete="fname"
                name="signup-userName"
                variant="outlined"
                fullWidth
                id="signup-firstName"
                label={t("username")}
                autoFocus
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                id="signup-email"
                label={t("emailaddress")}
                name="signup-email"
                autoComplete="email"
              />
            </Grid>
            <Grid item xs={12}>
              <FormControl variant="outlined" className={classes.formControl}>
                <InputLabel id="demo-simple-select-outlined-label">
                  {t("role_label")}
                </InputLabel>
                <Select
                  name="role"
                  labelId="demo-simple-select-outlined-label"
                  id="demo-simple-select-outlined"
                  value={role}
                  onChange={onchange}
                  label={t("role_label")}
                >
                  {userRole === "SuperAdmin" && (
                    <MenuItem value="Admin">{t("admin")}</MenuItem>
                  )}
                  <MenuItem value="User">{t("user")}</MenuItem>
                </Select>
              </FormControl>
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="signup-password"
                label={t("password")}
                type="password"
                id="signup-password"
                autoComplete="new-password"
                // inputProps={{
                //   maxLength: 12,
                //   minLength: 8,
                // }}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                fullWidth
                onChange={onchange}
                name="signup-confirmPassword"
                label={t("confirmpassword_label")}
                type="password"
                id="signup-confirmPassword"
                autoComplete="new-password"
                // inputProps={{
                //   maxLength: 12,
                //   minLength: 8
                // }}
              />
            </Grid>
            {/* <Grid item xs={12}>
              <FormControlLabel
                control={<Checkbox value="allowExtraEmails" color="primary" />}
                label="Remember me."
              />
            </Grid> */}
          </Grid>
          {loading && <LinearProgress color="secondary" />}
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            {t("signup")}
          </Button>
        </form>
      </div>
      <Notification
        id="notification"
        open={open} // must be a "state" to show notification message [true or false]
        setOpen={setOpen} // this function changes the state "open" value.
        error={error} // if error message put [true] else for success put [false]
        errorMessage={errorMessage} // the recived error message (array of string) from server.
      />
      <Box mt={5}>
        <Copyright />
      </Box>
    </Container>
  );
}
