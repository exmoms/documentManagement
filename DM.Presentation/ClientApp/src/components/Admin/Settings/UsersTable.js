import AddBox from "@material-ui/icons/AddBox";
import ArrowDownward from "@material-ui/icons/ArrowDownward";
import { default as Check } from "@material-ui/icons/Check";
import ChevronLeft from "@material-ui/icons/ChevronLeft";
import ChevronRight from "@material-ui/icons/ChevronRight";
import Clear from "@material-ui/icons/Clear";
import DeleteOutline from "@material-ui/icons/DeleteOutline";
import { default as Edit, default as EditIcon } from "@material-ui/icons/Edit";
import FilterList from "@material-ui/icons/FilterList";
import FirstPage from "@material-ui/icons/FirstPage";
import LastPage from "@material-ui/icons/LastPage";
import Remove from "@material-ui/icons/Remove";
import SaveAlt from "@material-ui/icons/SaveAlt";
import Search from "@material-ui/icons/Search";
import ViewColumn from "@material-ui/icons/ViewColumn";
import UpdateIcon from "@material-ui/icons/Update";
import MaterialTable from "material-table";
import React, { Component, forwardRef } from "react";
import { withTranslation } from "react-i18next";
import { fetchData } from "../../../api/FetchData";
import TableBodyRow from "../../Shared/materialTableRow";
import TableBodyToolbar from "../../Shared/materialTableToolbar";
import EditProfilePopup from "./EditProfilePopup";
import EditRolePopup from "./EditRolePopup";

const tableIcons = {
  Add: forwardRef((props, ref) => <AddBox {...props} ref={ref} />),
  Check: forwardRef((props, ref) => <Check {...props} ref={ref} />),
  Clear: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
  Delete: forwardRef((props, ref) => <DeleteOutline {...props} ref={ref} />),
  DetailPanel: forwardRef((props, ref) => (
    <ChevronRight {...props} ref={ref} />
  )),
  Edit: forwardRef((props, ref) => <Edit {...props} ref={ref} />),
  Export: forwardRef((props, ref) => <SaveAlt {...props} ref={ref} />),
  Filter: forwardRef((props, ref) => <FilterList {...props} ref={ref} />),
  FirstPage: forwardRef((props, ref) => <FirstPage {...props} ref={ref} />),
  LastPage: forwardRef((props, ref) => <LastPage {...props} ref={ref} />),
  NextPage: forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
  PreviousPage: forwardRef((props, ref) => (
    <ChevronLeft {...props} ref={ref} />
  )),
  ResetSearch: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
  Search: forwardRef((props, ref) => <Search {...props} ref={ref} />),
  SortArrow: forwardRef((props, ref) => <ArrowDownward {...props} ref={ref} />),
  ThirdStateCheck: forwardRef((props, ref) => <Remove {...props} ref={ref} />),
  ViewColumn: forwardRef((props, ref) => <ViewColumn {...props} ref={ref} />),
};

class UsersTable extends Component {
  getAllLessUsersURL = "/api/user/lessRoleUsers";
  initUser = {
    userId: 0,
    userName: "",
  };

  constructor(props) {
    super(props);
    this.state = {
      option: 1,
      data: [],
      user: this.initUser,
      show: false,
      showChangeRolePopup: false,
    };
  }

  componentDidMount() {
    fetchData(this.getAllLessUsersURL).then(
      (data) => {
        this.setState({
          data: data,
        });
      },
      (error) => console.log(error)
    );
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.userTableNeedUpdate &&
      this.props.userTableNeedUpdate !== prevProps.userTableNeedUpdate
    ) {
      fetchData(this.getAllLessUsersURL).then(
        (data) => {
          this.setState({
            data: data,
          });
        },
        (error) => console.log(error)
      );
    }
  }

  // translation object
  t = this.props["t"];

  editUserProfile = {
    icon: EditIcon,
    tooltip: this.t("edit_user_profile"),
    position: "row",
    onClick: (event, rowData) => {
      this.setState({ user: rowData, show: true });
    },
  };

  changeRole = {
    icon: UpdateIcon,
    tooltip: this.t("edit_user_role"),
    position: "row",
    onClick: (event, rowData) => {
      this.setState({ user: rowData, showChangeRolePopup: true });
    },
  };

  composeTableColumns = () => {
    let columns = [];

    columns = [
      {
        title: this.t("userid"),
        field: "userId",
      },
      { title: this.t("username"), field: "userName" },
      {
        title: this.t("role_label"),
        field: "role",
        hidden: true,
      },
    ];
    return columns;
  };

  composeTableActions = () => {
    let actions = [];
    let userRole = localStorage.getItem("userRole");
    if (userRole !== null && userRole !== undefined) {
      actions =
        userRole === "SuperAdmin"
          ? [this.editUserProfile, this.changeRole]
          : [this.editUserProfile];
    } else {
      actions = [this.editUserProfile];
    }

    return actions;
  };

  showPopup = (show) => {
    this.setState({ show: show });
  };

  showChangeRolePopup = (show, flag) => {
    if (flag) {
      fetchData(this.getAllLessUsersURL).then(
        (data) => {
          this.setState({
            data: data,
            showChangeRolePopup: show,
          });
        },
        (error) => console.log(error)
      );
    } else {
      this.setState({ showChangeRolePopup: show });
    }
  };

  render() {
    const { user, show, showChangeRolePopup } = this.state;
    return (
      <div style={{ maxWidth: "100%" }}>
        <MaterialTable
          title={this.t("browse_users")}
          icons={tableIcons}
          columns={this.composeTableColumns()}
          components={{
            Row: TableBodyRow,
            Toolbar: TableBodyToolbar,
          }}
          data={this.state.data}
          actions={this.composeTableActions()}
          options={{
            actionsColumnIndex: -1,
            exportButton: false,
          }}
          localization={{
            pagination: {
              labelRowsSelect: this.t("rows"),
              firstAriaLabel: this.t("first_page"),
              firstTooltip: this.t("first_page"),
              previousAriaLabel: this.t("previous_page"),
              previousTooltip: this.t("previous_page"),
              nextAriaLabel: this.t("next_page"),
              nextTooltip: this.t("next_page"),
              lastAriaLabel: this.t("last_page"),
              lastTooltip: this.t("last_page"),
            },
            toolbar: {
              nRowsSelected: "{0}" + this.t("selected_row"),
              searchTooltip: this.t("search"),
              searchPlaceholder: this.t("search"),
              exportTitle: this.t("export_as_csv"),
            },
            header: {
              actions: this.t("actions"),
            },
            body: {
              emptyDataSourceMessage: this.t("no_records"),
            },
          }}
        />
        <EditProfilePopup show={show} action={this.showPopup} user={user} />
        <EditRolePopup
          show={showChangeRolePopup}
          action={this.showChangeRolePopup}
          user={user}
        />
      </div>
    );
  }
}

export default withTranslation()(UsersTable);
