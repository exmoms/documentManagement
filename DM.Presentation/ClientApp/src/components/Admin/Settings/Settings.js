/**
 * @file Contains Settings component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import ExpansionPanel from "@material-ui/core/ExpansionPanel";
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import React, { Component } from "react";
import ConfigEmail from "./ConfigEmail";
import SignUp from "./SingUp";
import UsersTable from "./UsersTable";
import { withTranslation } from "react-i18next";

const useStyles = (theme) => ({
  root: {
    width: "100%",
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
});

class Settings extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userTableNeedUpdate: false,
    };
  }

  componentDidUpdate() {
    if (this.state.userTableNeedUpdate) {
      this.setState({ userTableNeedUpdate: false });
    }
  }

  updateUserTable = (value) => {
    this.setState({ userTableNeedUpdate: value });
  };

  t = this.props["t"];
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <ExpansionPanel>
          <ExpansionPanelSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls="panel1a-content"
            id="panel1a-header"
          >
            <Typography className={classes.heading}>
              {this.t("add_new_user")}
            </Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails>
            <SignUp updateUserTable={this.updateUserTable} />
          </ExpansionPanelDetails>
        </ExpansionPanel>
        <ExpansionPanel>
          <ExpansionPanelSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls="panel2a-content"
            id="panel2a-header"
          >
            <Typography className={classes.heading}>
              {this.t("edit_user_profile")}
            </Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails>
            <div style={{ margin: "auto" }}>
              <UsersTable
                userTableNeedUpdate={this.state.userTableNeedUpdate}
              />
            </div>
          </ExpansionPanelDetails>
        </ExpansionPanel>
        <ExpansionPanel>
          <ExpansionPanelSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls="panel3a-content"
            id="panel3a-header"
          >
            <Typography className={classes.heading}>
              {this.t("configure_email")}
            </Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails>
            <ConfigEmail />
          </ExpansionPanelDetails>
        </ExpansionPanel>
      </div>
    );
  }
}

export default withStyles(useStyles)(withTranslation()(Settings));
