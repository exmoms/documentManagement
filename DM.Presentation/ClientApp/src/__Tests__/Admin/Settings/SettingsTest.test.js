import React from "react";
import ReactDom from "react-dom";
import Settings from "../../../components/Admin/Settings/Settings";
import ConfigEmail from "../../../components/Admin/Settings/ConfigEmail";
import SingUp from "../../../components/Admin/Settings/SingUp";
import UsersTable from "../../../components/Admin/Settings/UsersTable";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
afterEach(cleanup);
describe("Settings", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(
        <Settings />,
      div
    );
  });

  test("matches snapshot", () => {
    const wrapper = shallow(
        <Settings />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  test("intialization ProfilePopUpByAdmin", () => {
  let  wrapper = mount(<Settings /> , { attachTo: document.body }) ;
  expect(wrapper.find(ConfigEmail).length).toBe(1);
  expect(wrapper.find(SingUp).length).toBe(1);
  expect(wrapper.find(UsersTable).length).toBe(1);
});
});