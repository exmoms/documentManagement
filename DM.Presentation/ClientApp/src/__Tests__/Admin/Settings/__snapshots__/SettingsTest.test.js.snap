// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
<withI18nextTranslation(Settings)
  classes={
    Object {
      "heading": "withI18nextTranslation(Settings)-heading-2",
      "root": "withI18nextTranslation(Settings)-root-1",
    }
  }
/>
`;
